courses/ocp4_advanced_deployment/lab_03_1/solve_lab.yml

```diff
@@ -523,47 +531,85 @@
       assert:
         that: r_nodes.stdout_lines | length == 2
 
+#
+## Switch Image Registry from PVC RWO Cinder to PVC RWX NFS
+#
+
+    - name: Wait until wrong PVC exists as installed by the installer
+      k8s_facts:
+        kind: PersistentVolumeClaim
+        namespace: openshift-image-registry
+      register: r_pvcs
+      retries: 20
+      delay: 30
+      until: r_pvcs.resources | length == 1
+      environment:
+        KUBECONFIG: "{{ lookup('env', 'HOME') }}/openstack-upi/auth/kubeconfig"
+      # tags: test_csv
+
+    - name: Delete registry PVC on cinder
+      k8s:
+        state: absent
+        api_version: v1
+        kind: PersistentVolumeClaim
+        name: image-registry-storage
+        namespace: openshift-image-registry
+      tags: test_csv
```

Comment:
I provision the environment for the first time, then I executed this part and see the following:

`
TASK [Delete registry PVC on cinder] *********************************************************************************************************************************************************************************************************
Sunday 19 July 2020  15:31:02 -0400 (0:00:01.859)       0:26:11.258 *********** 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: argument of type 'NoneType' is not iterable
fatal: [localhost]: FAILED! => changed=false 
  module_stderr: |-
    Traceback (most recent call last):
      File "/tmp/ansible_k8s_payload_tg6dc4bd/ansible_k8s_payload.zip/ansible/module_utils/k8s/common.py", line 184, in get_api_client
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/incluster_config.py", line 96, in load_incluster_config
        cert_filename=SERVICE_CERT_FILENAME).load_and_set()
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/incluster_config.py", line 47, in load_and_set
        self._load_config()
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/incluster_config.py", line 53, in _load_config
        raise ConfigException("Service host/port is not set.")
    kubernetes.config.config_exception.ConfigException: Service host/port is not set.
  
    During handling of the above exception, another exception occurred:
  
    Traceback (most recent call last):
      File "/home/gmontalv-redhat.com/.ansible/tmp/ansible-tmp-1595187062.51-150631367198791/AnsiballZ_k8s.py", line 114, in <module>
        _ansiballz_main()
      File "/home/gmontalv-redhat.com/.ansible/tmp/ansible-tmp-1595187062.51-150631367198791/AnsiballZ_k8s.py", line 106, in _ansiballz_main
        invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
      File "/home/gmontalv-redhat.com/.ansible/tmp/ansible-tmp-1595187062.51-150631367198791/AnsiballZ_k8s.py", line 49, in invoke_module
        imp.load_module('__main__', mod, module, MOD_DESC)
      File "/usr/lib64/python3.6/imp.py", line 235, in load_module
        return load_source(name, filename, file)
      File "/usr/lib64/python3.6/imp.py", line 170, in load_source
        module = _exec(spec, sys.modules[name])
      File "<frozen importlib._bootstrap>", line 618, in _exec
      File "<frozen importlib._bootstrap_external>", line 678, in exec_module
      File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
      File "/tmp/ansible_k8s_payload_tg6dc4bd/__main__.py", line 258, in <module>
      File "/tmp/ansible_k8s_payload_tg6dc4bd/__main__.py", line 254, in main
      File "/tmp/ansible_k8s_payload_tg6dc4bd/ansible_k8s_payload.zip/ansible/module_utils/k8s/raw.py", line 154, in execute_module
      File "/tmp/ansible_k8s_payload_tg6dc4bd/ansible_k8s_payload.zip/ansible/module_utils/k8s/common.py", line 186, in get_api_client
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 645, in load_kube_config
        persist_config=persist_config)
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 613, in _get_kube_config_loader_for_yaml_file
        **kwargs)
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 153, in __init__
        self.set_active_context(active_context)
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 173, in set_active_context
        context_name = self._config['current-context']
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 495, in __getitem__
        v = self.safe_get(key)
      File "/usr/local/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 491, in safe_get
        key in self.value):
    TypeError: argument of type 'NoneType' is not iterable
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=70   changed=50   unreachable=0    failed=1    skipped=0    rescued=0    ignored=4   
utilityvm.example.com      : ok=21   changed=15   unreachable=0    failed=0    skipped=17   rescued=0    ignored=1   
`

Then I execute the solver again and it works.